packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -66,49 +66,71 @@ const getProvider = async (state?: Partial<MetadataProviderCredentials>) => {
     return providerInstance;
 };
 
-export const enableMetadata = () => (dispatch: Dispatch) => {
-    dispatch({
-        type: METADATA.ENABLE,
-    });
-};
-
-export const disposeMetadata = () => (dispatch: Dispatch, getState: GetState) => {
+/**
+ * dispose metadata from all labelable objects.
+ * Has keys parameter,
+ * if true - dispose also metadata keys
+ * if false - dispose only metadata values
+ */
+export const disposeMetadata = (keys?: boolean) => (dispatch: Dispatch, getState: GetState) => {
     getState().wallet.accounts.forEach(account => {
+        const updatedMetadata = { ...account.metadata };
+
+        // always remove metadata  values
+        updatedMetadata.outputLabels = {};
+        updatedMetadata.addressLabels = {};
+        updatedMetadata.accountLabel = '';
+
+        // and sometimes remove also keys (information we can only if device is connected)
+        if (keys) {
+            updatedMetadata.fileName = '';
+            updatedMetadata.aesKey = '';
+        }
+
         dispatch({
             type: METADATA.ACCOUNT_ADD,
             payload: {
                 ...account,
-                metadata: {
-                    key: account.metadata.key,
-                    fileName: '',
-                    aesKey: '',
-                    outputLabels: {},
-                    addressLabels: {},
-                    accountLabel: '',
-                },
+                metadata: updatedMetadata,
             },
         });
     });
+
     getState().devices.forEach(device => {
         if (device.state) {
+            let updatedMetadata = { ...device.metadata };
+
+            if (keys) {
+                // set metadata as disabled for this device, remove all metadata related information
+                updatedMetadata = { status: 'disabled' };
+            } else if ('key' in updatedMetadata) {
```

Comment:
Very minor: `in` works here but it's inconsistent with how we test for existing properties across the project. Or maybe I'm missing a specific use-case for this operator?

Reply:
Using it as narrowing expression for types here. This metadata object may have 2 different shape where one of them dos not have 'key' property. So trying to access it in plain javascript would result in typescript error here.